Common/src/Common/Service/Table/TableBuilder.php

```diff
@@ -1710,10 +1710,7 @@ private function callFormatter($column, $data)
 
             // Check if the formatter class contains a namespace
             if (strpos($formatterClass, '\\') === false) {
-                @trigger_error(sprintf('Table formatter "%s" should be using the FQCN.', $column['formatter']), \E_USER_DEPRECATED);
-
-                // Append the namespace if it's missing
-                $formatterClass = '\\' . __NAMESPACE__ . '\\Formatter\\' . $formatterClass;
+                @trigger_error(sprintf('Table formatter "%s" should be using the FQCN.', $column['formatter']), \E_ERROR);
             }
```

Comment:
Would it be a good idea to remove this conditional and let it be caught by the next conditional (line 1716) instead?

A non-FQCN will not be found in the service container (plugin manager in this instance) and will throw the `MissingFormatterException`.

Reply:
Would this be harder for anyone using it to debug? If they have a non-FQCN, they aren't missing a table formatter but the error message tells them they are.

Reply:
I don't believe so - though it is a good consideration to raise.

Ultimately a missing class from the service container and a non-FQCN parameter mean the same though - we're asking the service container if a key exists - FQCN or a non-FQCN, both are strings that are passed to the service container.

If you're not persuaded, feel free to leave it in - but we do need to throw an exception rather than using a suppressed (using `@` - [source](https://www.php.net/manual/en/language.operators.errorcontrol.php)) `trigger_error` as it doesn't halt the script execution. We didn't need to halt script execution for deprecation errors so was fine to suppress.

Reply:
I understand that non-FQCN and a missing class will be treated the same. My first idea was to swap the conditional statements around, but that would just change the order in which the error messages come out.

Instead, I have removed the conditional: `if (strpos($formatterClass, '') === false)`, but I altered the error message to try to be more descriptive of the issue they might have. However, this does cause the test to fail as it's not expecting the extra detail. Would I be allowed to change this? It is on line 2705 of TableBuilderTest.php.